### DISCUSSION TITLE
SSH From Windows Not Working

### DISCUSSION THREAD

**Discussion by swartjie** [spatie/ssh#84] [Q&A]
Hey Guys,

I am trying to SSH to a linux box from my windows pc.
Using the pvt key works in a ssh terminal, but through the below example, I am getting a general error...

How can I get it to work from a windows pc for development?

$process = Ssh::create('root', 'xx.xx.xx.xx', 22)
            ->usePrivateKey(storage_path('id_rsa'))
            ->execute([
                'mkdir hello',
                'ls -alh',
            ]);

**Comment by pniaps** [comment]
I changed the function **getExecuteCommand** to return this and works

```
return "ssh {$extraOptions} {$target} {$commandString}";
```

but its only a test. `$commandString` would need to be sanitized, checked, etc...

**Comment by PerryRylance** [comment]
I ended up forking this and adding a shim in getExecuteCommand, it's on Packagist so if you want to use it then `composer require perry-rylance/ssh`

Source is over at https://github.com/PerryRylance/ssh
